app/assets/scripts/views/field-report-form.js

```diff
@@ -297,18 +260,21 @@ class FieldReportForm extends React.Component {
   }
 
   componentWillReceiveProps (nextProps) {
-    if (this.fieldReportForm.fetching && !nextProps.fieldReportForm.fetching) {
-      if (!nextProps.fieldReportForm) {
-        console.log('nextProps.fieldReportForm', nextProps.fieldReportForm);
-        hideGlobalLoading();
+    if (this.props.fieldReportForm.fetching && !nextProps.fieldReportForm.fetching) {
+      hideGlobalLoading();
+      if (nextProps.fieldReportForm.error) {
+        const message = nextProps.fieldReportForm.error.error_message || 'Could not submit field report';
+        showAlert('danger', <p><strong>Error:</strong> {message}</p>, true, 4500);
+      } else {
+        const { id } = nextProps.fieldReportForm.data;
+        showAlert('success', <p><Link to={`/reports/${id}`}>Field report created</Link></p>, true, 9000);
```

Comment:
Maybe this alert shouldn't autodismiss?

Reply:
@danielfdsilva actually I think this should redirect to the field report page. If you click save again it will create a duplicate record, which is not great.